src/torcontrol.cpp

```diff
@@ -752,7 +753,7 @@ void StartTorControl()
         return;
     }
 
-    torControlThread = std::thread(std::bind(&TraceThread<void (*)()>, "torcontrol", &TorControlThread));
+    torControlThread = std::thread(&TraceThread<std::function<void()> >, "torcontrol", [target, listen_port] { TorControlThread(target, listen_port); });
```

Comment:
Do you want to capture `target` and `listen_port` by reference?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/14856#discussion_r271595261

> Do you want to capture target and listen_port by reference?

Definitely not `listen_port` because it will go out of scope. Probably not `target` because it is a reference to an object this has no control over. In C++14, the better thing would be to pass `target` by value then capture-by-move. But the current code is reasonable and the simplest correct implementation for C++11.